.github/JAVASCRIPT_CODESTYLE.md

```diff
@@ -0,0 +1,563 @@
+
+*Right:*
+
+```js
+if (true) {
+  console.log('winning');
+}
+```
+
+*Wrong:*
+
+```js
+if (true)
+{
+  console.log('losing');
+}
+```
+
+Also, notice the use of whitespace before and after the condition statement.
+
+### Declare one variable per var statement
+
+Declare one variable per var statement, it makes it easier to re-order the
+lines. However, ignore [Crockford][crockfordconvention] when it comes to
+declaring variables deeper inside a function, just put the declarations wherever
+they make sense.
+
+*Right:*
+
+```js
+var keys   = ['foo', 'bar'];
+var values = [23, 42];
+
+var object = {};
+while (keys.length) {
+  var key = keys.pop();
+  object[key] = values.pop();
+}
+```
+
+*Wrong:*
+
+```js
+var keys = ['foo', 'bar'],
+    values = [23, 42],
+    object = {},
+    key;
+
+while (keys.length) {
+  key = keys.pop();
+  object[key] = values.pop();
+}
+```
+
+[crockfordconvention]: http://javascript.crockford.com/code.html
+
+### Naming Conventions
+
+### Use lowerCamelCase for variables, properties and function names
+
+Variables, properties and function names should use `lowerCamelCase`.  They
+should also be descriptive. Single character variables and uncommon
+abbreviations should generally be avoided.
+
+*Right:*
+
+```js
+var adminUser = db.query('SELECT * FROM users ...');
+```
+
+*Wrong:*
+
+```js
+var admin_user = db.query('SELECT * FROM users ...');
+```
+
+### Use UpperCamelCase for class names
+
+Class names should be capitalized using `UpperCamelCase`.
+
+*Right:*
+
+```js
+function BankAccount() {
+}
+```
+
+*Wrong:*
+
+```js
+function bank_Account() {
+}
+```
+
+## Use UPPERCASE for Constants
+
+Constants should be declared as regular variables or static class properties,
+using all uppercase letters.
+
+*Right:*
+
+```js
+var SECOND = 1 * 1000;
+
+function File() {
+}
+File.FULL_PERMISSIONS = 0777;
+```
+
+*Wrong:*
+
+```js
+const SECOND = 1 * 1000;
+
+function File() {
+}
+File.fullPermissions = 0777;
+```
+
+[const]: https://developer.mozilla.org/en/JavaScript/Reference/Statements/const
+
+## Variables
+
+### Object / Array creation
+
+Use trailing commas and put *short* declarations on a single line. Only quote
+keys when your interpreter complains:
+
+*Right:*
+
+```js
+var a = ['hello', 'world'];
+var b = {
+  good: 'code',
+  'is generally': 'pretty',
+};
+```
+
+*Wrong:*
+
+```js
+var a = [
+  'hello', 'world'
+];
+var b = {"good": 'code'
+        , is generally: 'pretty'
+        };
+```
+
+## Conditionals
+
+### Use the === operator
+
+Programming is not about remembering [stupid rules][comparisonoperators]. Use
+the triple equality operator as it will work just as expected.
+
+*Right:*
+
+```js
+var a = 0;
+if (a !== '') {
+  console.log('winning');
+}
+
+```
+
+*Wrong:*
+
+```js
+var a = 0;
+if (a == '') {
+  console.log('losing');
+}
+```
+
+[comparisonoperators]: https://developer.mozilla.org/en/JavaScript/Reference/Operators/Comparison_Operators
+
+### Use multi-line ternary operator
```

Comment:
This would disagree with our vendor files and some files of our own. We use inline ternary operators, not to be difficult, but because everyone uses it that way.